QSB/WorldRegistry.cs

```diff
@@ -0,0 +1,39 @@
+﻿using QSB.Events;
+using QSB.Utility;
+using QSB.WorldSync;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace QSB
+{
+    public static class WorldRegistry
+    {
+        public static List<GeyserController> GeyserControllers = new List<GeyserController>();
+
+        public static void GenerateComponentList()
+        {
+            GeyserControllers = Resources.FindObjectsOfTypeAll<GeyserController>().ToList();
+
+            foreach (var component in GeyserControllers)
+            {
+                if (NetworkServer.active)
+                {
+                    component.OnGeyserActivateEvent += () => GlobalMessenger<GeyserController, bool>.FireEvent(EventNames.QSBGeyserState, component, true);
+                    component.OnGeyserDeactivateEvent += () => GlobalMessenger<GeyserController, bool>.FireEvent(EventNames.QSBGeyserState, component, false);
+                }
+            }
+        }
+
+        public static int GetObjectID(SyncObjects type, object component)
```

Comment:
Suggestion: we make an abstract class called WorldObject, then you make a wrapper for GeyserController which inherits from WorldObject, and ElevatorController also inherits from WorldObject. The list in WorldRegistry would contain world objects. That would help clean up WorldRegistry a lot